jwst/ramp_fitting/ramp_fit.py

```diff
@@ -2940,12 +2944,11 @@ def calc_num_seg(gdq, n_int):
     """
     max_cr = 0  # max number of CRS for all integrations
 
-    # For all 2d pixels, get max number of CRs along their ramps
-    # to use as a surrogate for the number of semiramps along the ramps
-    for nint in range(n_int):  # loop over integrations
-        gdq_cr = np.bitwise_and( gdq[nint, :, :, :], dqflags.group['JUMP_DET'])
-        temp_max_cr = int((gdq_cr.sum(axis=0)).max()/dqflags.group['JUMP_DET'])
-        max_cr = max( max_cr, temp_max_cr )
+    # For all 2d pixels, get max number of CRs or DO_NOT_USE flags along their
+    # ramps, to use as a surrogate for the number of segments along the ramps
+    # Note that we only care about flags that are NOT in the first or last groups,
+    # because exclusion of a first or last group won't result in an additional segment.
+    max_cr = np.count_nonzero(np.bitwise_and(gdq[:, 1:-1], JUMP_DET | DO_NOT_USE), axis=1).max()
```

Comment:
note that max_cr is bumped up by 1 later in line 2956, to account for the fact that n flags implies n+1 segments.

Reply:
Minor comment: Is there a reason not to bump now? Or is the later use a conceptually different value, segments, as opposed to "maximum rejects".

Reply:
No really good reason I suppose, although they are a bit different conceptually. This line simply computes the max number "interrupts" in any ramp, and then that number of interrupts turns into n+1 segments for fitting. Besides, the line is so long already, I didn't want to add yet another operation on the end of it!